src/geo/leaflet/leaflet-cartodb-vector-layer-group-view.js

```diff
@@ -27,6 +27,19 @@ var LeafletCartoDBVectorLayerGroupView = CartoDBd3Layer.extend({
       this.setCartoCSS(index, style);
     }, this);
 
+    layerModel.layers.bind('change:meta', function (child, meta) {
+      if (this.model.get('type') === 'namedmap') {
```

Comment:
Why only for named maps? It would be nice if we didn't have to check this and both named and anonymous maps worked the same way (i.e: set the URL to the provider once all styles have been (re)seted?)